06_neighborhoods/Transportation/transit_trips_cost_city.R

```diff
@@ -0,0 +1,292 @@
+                            pattern="*.csv")
+print(tracts19files)
+tractpath19 = file.path("C:/Users/USERNAME/Box/Lab/Projects/Gates Upward Mobility Framework/Outreach and Tools/Data/Metrics_2024_round/Transportation/2019_tract",tracts19files)
+print(tractpath19)
+transport_tracts_2019 <- map_df(tractpath19, read_csv)
+
+# create correct FIPS columns
+transport_tracts_2019 <- transport_tracts_2019 %>%
+  rename (GEOID = tract) %>%
+  mutate(
+    state = substr(GEOID, start = 2, stop = 3),
+    county = substr(GEOID, start = 4, stop = 6),
+    tract = substr(GEOID, start = 7, stop = 12)
+  )
+
+# keep only variables of interest
+transit_trips_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, transit_trips_80ami)
+transit_cost_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, t_80ami)
+
+
+###################################################################
+
+# 2. Crosswalk to the cities (from tracts) & aggregate data
+
+# bring in the crosswalk of interest (we want the year range to be before 2020, when tract definitions changed)
+tract_place <- read_csv("geographic-crosswalks/data/geocorr2018_tract_to_place.csv")
+# remove the decimal from the tract variable
+tract_place <- tract_place %>%
+  mutate(tract = tract*100)
+# add in leading zeroes as needed
+tract_place <- tract_place %>%
+  mutate(state = sprintf("%0.2d", as.numeric(state)),
+         tract = sprintf("%0.6d", as.integer(as.character(tract))),
+         placefp = sprintf("%0.5d", as.numeric(placefp)),
+         county = substr(county, nchar(county) - 2, nchar(county))
+  ) %>%
+  rename(place = placefp)
+
+
+# NOW, aggregate the tract-level data to the place levels for each variable for each year
+
+# For when we want to: Limit to the Census Places we want 
+# first, bring in the places crosswalk (place-populations.csv)
+places <- read_csv("geographic-crosswalks/data/place-populations.csv")
+# keep only the most relevant year
+places <- places %>%
+  filter(year == 2019)
+
+
+# TRANSPORTATION COST 2015
+transit_cost_city_2015 <- left_join(transit_cost_tracts_2015, tract_place, by=c("state", "county", "tract"))
+# collapse to places and also create data quality marker
+# data quality can be 1 when most of the tracts that fall in the place (e.g., >50% of the tracts) 
+# have most of their area falling in the place (e.g., >50% of the tract's area is in the place)
+# otherwise, data quality is 2
+transit_cost_city_2015 <- transit_cost_city_2015 %>% 
+  dplyr::group_by(state, place) %>% 
+  dplyr::summarize(index_transportation_cost = round(mean(t_80ami), 2),
+                   n = n(),
+                   dq = sum(afact > 0.5, na.rm = TRUE)
+  )%>%
+  dplyr::mutate(index_transportation_cost_quality = ifelse(dq / n > 0.5, 1, ifelse(dq / n <= 0.5, 2, NA)))
```

Comment:
You are using `afact` here to create the quality score, which I think is OK if that's how you want to define it, but I still think the mean itself needs to be weighted by `afact`.

Reply:
Agreed, adjusted